components/servo/lib.rs

```diff
@@ -998,6 +999,28 @@ impl Servo {
                     webview.delegate().show_form_control(webview, form_control);
                 }
             },
+            EmbedderMsg::GetWindowRect(webview_id, response_sender) => {
+                let window_rect = if let Some(webview) = self.get_webview_handle(webview_id) {
+                    let dpi = webview.dpi();
+                    let screen = webview.delegate().screen_geometry(webview).unwrap();
+                    let pos = screen.window_offset.to_f32();
+                    let window_size = screen.window_size.to_f32();
+
+                    let pos_in_css_pixel = (pos / dpi).round().to_i32();
+                    let window_size_in_css_pixel = (window_size / dpi).round().to_i32();
+                    DeviceIndependentIntRect::from_origin_and_size(
+                        pos_in_css_pixel,
+                        window_size_in_css_pixel,
+                    )
+                } else {
+                    warn!("{webview_id} is invalid");
```

Comment:
I think you can omit the warning since we aren't doing this consistently in this file.